### PR TITLE
EVG-5560 support module updates in patches

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: service/api_plugin_manifest.go [hunk 1]
```diff
@@ -73,3 +74,25 @@ func (as *APIServer) manifestLoadHandler(w http.ResponseWriter, r *http.Request)
...
+	for _, module := range modules {
+		owner, repo := module.GetRepoOwnerAndName()
+		projectModules[module.Name] = manifest.Module{
+			Branch: module.Branch,
+			Repo:   repo,
+			Owner:  owner,
+		}
+	}
+
+	return reflect.DeepEqual(manifestModules, projectModules)
```
Comment (tychoish) [inline service/api_plugin_manifest.go]: I'd implement this as a method on a type alias for []model.Module, and probably do it without reflection (make them sortable, call sort, and iterate through one, and return false if there's a difference.

File: model/manifest/db.go [hunk 2]
```diff
@@ -52,3 +53,17 @@ func ByProjectAndRevision(project, revision string) db.Q {
 		ManifestRevisionKey: revision,
 	})
 }
+
+func GetVersionManifest(versionID, project, revision string) (*Manifest, error) {
```
[the remark this answers is not in the document]
Comment (john-m-liu) [reply model/manifest/db.go]: renamed to FindFromVersion

File: service/api_plugin_manifest.go (hunk 1 above)
Comment (john-m-liu) [reply service/api_plugin_manifest.go]: Made an alias on []Module but I like the reflection implementation better. I think it's quite reasonable and I don't see any reason to duplicate this

File: service/api_plugin_manifest.go [hunk 3]
```diff
@@ -47,13 +47,13 @@ func (as *APIServer) manifestLoadHandler(w http.ResponseWriter, r *http.Request)
...
 	}
-	currentManifest, err := manifest.FindOne(manifest.ByProjectAndRevision(v.Identifier, v.Revision))
+	currentManifest, err := manifest.FindFromVersion(v.Id, v.Identifier, v.Revision)
 	if err != nil {
 		as.LoggedError(w, r, http.StatusBadRequest,
 			errors.Wrapf(err, "error retrieving manifest with version id %s", task.Version))
 		return
 	}
-	if currentManifest != nil {
+	if currentManifest != nil && project.Modules.IsIdentical(*currentManifest) {
```
[the remark this answers is not in the document]
Comment (john-m-liu) [reply service/api_plugin_manifest.go]: The scenario you're describing shouldn't be relevant here. If a user modifies the modules section of the project config, they are expecting that they are modifying the entire list of modules. If they are using the patch-set-module command, that sets the module directly in the patch and does not affect the manifest.

Though while looking at this it appears that the scenario you're describing has never worked correctly. I'm seeing that we always build the manifest from the list of modules in the evergreen.yml committed to the repo, so modifying this in a patch has never done anything. Have you tried adding or changing the branch of a module in a patch and had it work?

[the remark this answers is not in the document]
Comment (visemet) [reply service/api_plugin_manifest.go]: > If a user modifies the modules section of the project config, they are expecting that they are modifying the entire list of modules.

I'm not sure that would be my expectation when I make a change to add a single entry to a list of modules.

> Have you tried adding or changing the branch of a module in a patch and had it work?

It definitely used to when patch builds generated their own manifest all the time.

**Pull request merged**